packages/browser-wallet/src/background/startup.ts

```diff
@@ -0,0 +1,87 @@
+import { HttpProvider, JsonRpcClient } from '@concordium/web-sdk';
+import { storedCurrentNetwork, storedCredentials, storedIdentities } from '@shared/storage/access';
+
+import {
+    Identity,
+    CreationStatus,
+    PendingIdentity,
+    PendingWalletCredential,
+    WalletCredential,
+} from '@shared/storage/types';
+import { IdentityTokenContainer, IdentityProviderIdentityStatus } from 'wallet-common-helpers/lib/utils/identity/types';
+import { updateCredentials, updateIdentities } from './update';
+
+const isPendingCred = (cred: WalletCredential): cred is PendingWalletCredential =>
+    cred.status === CreationStatus.Pending;
+const isPendingIdentity = (identity: Identity): identity is PendingIdentity =>
+    identity.status === CreationStatus.Pending;
+const updateInterval = 10000;
+
+/**
+ * Continously checks whether pending credentials have been confirmed.
+ */
+async function monitorAccountStatus() {
+    setTimeout(async function loop() {
+        const network = await storedCurrentNetwork.get();
+        const url = network?.jsonRpcUrl;
+        const creds = await storedCredentials.get();
+        const toUpdate: WalletCredential[] = [];
+        if (url && creds) {
+            const client = new JsonRpcClient(new HttpProvider(url, fetch));
+            for (const { status, deploymentHash, ...info } of creds.filter(isPendingCred)) {
+                const response = await client.getTransactionStatus(deploymentHash);
+                if (response?.status === 'finalized') {
+                    const isSuccessful = Object.values(response?.outcomes || {}).some(
+                        (outcome) => outcome.result.outcome === 'success'
+                    );
+                    toUpdate.push({
+                        ...info,
+                        status: isSuccessful ? CreationStatus.Confirmed : CreationStatus.Rejected,
+                    });
+                }
+            }
+            if (toUpdate.length) {
+                await updateCredentials(toUpdate);
+            }
+        }
+        setTimeout(loop, updateInterval);
+    }, 0);
+}
+
+/**
+ * Continously checks whether pending identities have been confirmed or rejected.
+ */
+async function monitorIdentityStatus() {
+    setTimeout(async function loop() {
```

Comment:
This does not seem like an ideal solution. We just have an infinite loop going on, instead of just checking the status on-demand. Can you make it so that the front-end pushes a message that ask for listening to the status instead? The only at startup do we need to read all stored identities to start listening for the pending ones, and the UI will then ask when it creates a new identity.

Reply:
If it seems like a bigger thing to refactor, then I can live with the loops initially and then we can re-do it later.

Reply:
I'll fix it later.